packages/react-components/src/components/chart/baseChart.tsx

```diff
@@ -1,36 +1,92 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { useECharts } from '../../hooks/useECharts';
 import { ChartOptions } from './types';
 import { useVisualizedDataStreams } from './useVisualizedDataStreams';
 import { convertOptions } from './converters/convertOptions';
+import { EChartsType, getInstanceByDom, SeriesOption, YAXisComponentOption } from 'echarts';
+import { convertYAxis } from './converters/convertAxis';
+import { convertSeriesAndYAxis, reduceSeriesAndYAxis } from './converters/convertSeriesAndYAxis';
+import { HotKeys, KeyMap } from 'react-hotkeys';
+import { addNewGraphic } from './utils/getInfo';
+
+const keyMap: KeyMap = {
+  commandDown: { sequence: 'command', action: 'keydown' },
+  commandUp: { sequence: 'command', action: 'keyup' },
+};
 
 /**
  * Base chart to display Line, Scatter, and Bar charts.
  */
 const Chart = ({ viewport, queries, size, ...options }: ChartOptions) => {
-  const { hasError, isLoading, dataStreams } = useVisualizedDataStreams(queries, viewport);
-
-  const option = convertOptions(
-    {
-      dataStreams,
-      hasError,
-      isLoading,
-    },
-    {
-      viewport,
-      queries,
-      size,
-      ...options,
-    }
+  const { isLoading, dataStreams } = useVisualizedDataStreams(queries, viewport);
+  const { axis } = options;
+  const defaultSeries: SeriesOption[] = [];
+  const defaultYAxis: YAXisComponentOption[] = [convertYAxis(axis)];
+  // Need series data for the calculation of Y for rendering a circle at TS and series lines intersections
+  const { series, yAxis } = useMemo(
```

Comment:
does this still need to be pulled out of convertOptions? I don't see series being used in this component except to be passed into convertOptions

Reply:
in this CR, we don't need it, but we will need the series data in the next set of changes.

Reply:
Is it possible to derive the info you need from the datastreams? otherwise let's just merge the converted options with the series and yAxis. I think it's weird to pass converted options into the convertOptions function.

Reply:
dataStreams could possibly used, but then we will have to process it similar to what we do for series... would rather prefer merging the output of convertOptions with series and YAxis